src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -414,7 +414,20 @@ event::ptr primitive_inst::realloc_if_needed() {
         return ev;
     }
 
-    bool can_reuse_buffer = _outputs[0] && actual_layout.count() <= max_output_layout_size;
+    // If the node's output memory is used as next input memory for the same node,
+    // the buffer should be allocated newly every time to prevent data corruption
+    bool output_used_as_its_input_at_next_iter = false;
+    if (_impl_params->output_buffer_used_for_next_input != "") {
+        for (auto dep : _deps) {
```

Comment:
Shouldn't we set `output_used_as_its_input_at_next_iter` immediately if `_impl_params->output_buffer_used_for_next_input != ""`, rather than only if its dependencies are marked as well?

Reply:
I think we only need to force new memory for output memory only if the memory is to be used by same node. 
i.e., 
inmem -> concat -> outmem
where outmem is to be used as inmem in the next iter. 
In such case, inmem is to be same as outmem if the memory is not to be allocated newly, 
(e.g., becuase current shape is smaller than previous shape), so that the input data will be currupted.

Reply:
Agree, thanks for explanation!